primer/src/Primer/Action/Available.hs

```diff
@@ -586,7 +588,14 @@ options typeDefs defs cxt level def0 sel0 = \case
                 | Just t <- exprType e -> do
                     pure $
                       vcs <&> \vc@(vc', tc, td) ->
-                        (globalOpt' (t `eqType` valConType tc td vc') (valConName vc'), vc)
+                        -- Here we ignore the arity of the constructor
+                        -- and consider only the type to which the
+                        -- constructor belongs. This matches, e.g.,
+                        -- both @Zero@ and @Succ i@ to a node whose
+                        -- type is @Nat@; whereas using 'valConType',
+                        -- which eta-expands the value constructor's
+                        -- type, would only match @Zero@.
+                        (globalOpt' (t `eqType` mkTAppCon tc (TVar () . fst <$> astTypeDefParameters td)) (valConName vc'), vc)
```

Comment:
I think we can do better here. (See below)

Background: we only have regular types (i.e. not gadts), so for a type `D a b` with a constructor `C A[a,b]`, we have that `D S T  ∋ C A[S,T]` (with no restrictions on what `S` and `T` (the instantiations of the type's params) are). Thus we only need to check whether the constructor `vc` is a constructor of `D` (and that `D S T` is fully-applied, though this should be guarenteed by the fact that `D S T` is `t` in the code, and `t = exprType e`, thus of kind `KType`.)

Thus we need to:
- check that `t` is of the form "a type constructor applied to a bunch of things"
- extract the head `D`
- check whether `vc` is a constructor of `D`

I.e. something like (off the top of my head):
`Just tc == fmap fst (decomposeTAppCon t)`

Reply:
This is implemented in https://github.com/hackworthltd/primer/pull/1071